### PR TITLE
fix decimal money parser to accept leading zeros

### PR DESCRIPTION
**Pull request by frederikbosch**
includes test for leading zeros in all parsers

### CODE REVIEW

File: src/Parser/DecimalMoneyParser.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
  */
 final class DecimalMoneyParser implements MoneyParser
 {
-    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>0|[1-9]\d*)?\.?(?P<fraction>\d+)?$/';
+    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>[0-9]\d*)?\.?(?P<fraction>\d+)?$/';
```
Comment (PowerKiKi) [inline src/Parser/DecimalMoneyParser.php:27]: The regexp can be simplified this way:

```suggestion
    public const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>\d+)?\.?(?P<fraction>\d+)?$/';
```

File: tests/Parser/DecimalMoneyParserTest.php [hunk 2]
```diff
@@ -115,6 +115,8 @@ public function formattedMoneyExamples(): array
             ['9.999', 'USD', 2, 1000],
             ['9.99', 'USD', 2, 999],
             ['-9.99', 'USD', 2, -999],
+            ['000009.99', 'USD', 2, 999],
+            ['-000009.99', 'USD', 2, -999],
```
Comment (PowerKiKi) [inline tests/Parser/DecimalMoneyParserTest.php:119]: Just to be sure that we don't wreak havoc with a `ltrim` in the future:

```suggestion
            ['-000009.99', 'USD', 2, -999],
            ['000', 'USD', 2, 0],
```

File: tests/Parser/BitcoinMoneyParserTest.php [hunk 3]
```diff
@@ -76,6 +76,8 @@ public function bitcoinExamples(): array
             ["\xC9\x8399.", 9900],
             ["\xC9\x830", '0'],
             ["\xC9\x830.020000000", 2],
+            ["\xC9\x830000.020000000", 2],
+            ["-\xC9\x830000.020000000", -2],
```
Comment (PowerKiKi) [inline tests/Parser/BitcoinMoneyParserTest.php:80]: Just to be sure that we don't wreak havoc with a `ltrim` in the future:

```suggestion
            ["-\xC9\x830000.020000000", -2],
            ["-\xC9\x83000000", 0],
```

File: tests/Parser/IntlMoneyParserTest.php [hunk 4]
```diff
@@ -149,6 +149,8 @@ public function formattedMoneyExamples(): array
             ['$.99', 99],
             ['-$.99', -99],
             ['$99.', 9900],
+            ['$00099.', 9900],
+            ['-$00099.', -9900],
```
Comment (PowerKiKi) [inline tests/Parser/IntlMoneyParserTest.php:153]: Idem:

```suggestion
            ['-$00099.', -9900],
            ['$000', 0],
```

**Comment by frederikbosch** [comment]
Thanks @PowerKiKi

**Pull request merged**